Sources/BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -429,20 +407,22 @@ extension KeyringStore: BraveWalletKeyringServiceObserver {
   public func autoLockMinutesChanged() {
   }
 
-  public func selectedAccountChanged(_ coinType: BraveWallet.CoinType) {
-    walletService.setSelectedCoin(coinType)
+  public func selectedWalletAccountChanged(_ account: BraveWallet.AccountInfo) {
+    updateKeyringInfo()
+  }
+  
+  public func selectedDappAccountChanged(_ coin: BraveWallet.CoinType, account: BraveWallet.AccountInfo?) {
     updateKeyringInfo()
   }
 
   public func keyringCreated(_ keyringId: BraveWallet.KeyringId) {
     Task { @MainActor in
       let newKeyring = await keyringService.keyringInfo(keyringId)
-      if let newKeyringCoin = newKeyring.coin {
-        let selectedAccount = await keyringService.selectedAccount(newKeyringCoin)
-        // if the new Keyring doesn't have a selected account, select the first account
-        if selectedAccount == nil, let newAccount = newKeyring.accountInfos.first {
-          await keyringService.setSelectedAccount(newAccount.accountId)
-        }
+      let selectedAccount = await keyringService.allAccounts().selectedAccount
+      // if the new Keyring doesn't have a selected account, select the first account
+      // TODO: verify if still needed
+      if selectedAccount == nil, let newAccount = newKeyring.accountInfos.first {
+        await keyringService.setSelectedAccount(newAccount.accountId)
       }
```

Comment:
not sure if you have checked. this i guess is still theoretically possible since `keyringService.allAccount().selectedAccount` is optional? i think its safe to keep.

Reply:
Yeah I think safe to keep as a fallback, but I haven't double checked this piece yet. We might just need to verify importing an account, setting it as the selected account and then deleting that account. I'll reply in this thread after checking (likely tomorrow).

> There is always some account selected. Backend is not responsible for selecting some account after removal of selected one. Removed lots of UIs code trying to recover from "empty" selected account.
https://github.com/brave/brave-core/pull/18930

Reply:
Removing selected imported account works as before. 

https://github.com/brave/brave-ios/assets/5314553/f9283e93-7b5c-4327-9f48-f2c43f63325c

As for this `setSelectedAccount` I think it can remain for now as a fallback on core logic for Solana/Filecoin.